data-raw/R2R_TMH_habitat_inputs/cache_tmh_data.R

```diff
@@ -145,143 +146,210 @@ for(i in 1:length(watersheds)) {
   
 }
 
+################################################################################
+################################################################################
+# SPRING AND WINTER RUNS: extend past reservoirs/dams ---------------------
+################################################################################
+################################################################################
+
+all_existing_and_tmh_data_wr_sr <- readRDS('data-raw/R2R_TMH_habitat_inputs/all_habitat_data_for_tmh_inputs_spring_winter_run.rdata')
+
+all_hab_data_long <- all_existing_and_tmh_data_wr_sr |>
+  rename(spwn_acres_max = max_spawning_acres,
+         rear_acres_max = max_rearing_acres, 
+         flood_acres_max = max_floodplain_acres) |> 
+  pivot_longer(cols = c(spwn_acres_max:flood_acres_max), names_to = "metric") |>
+  separate(metric, c('hab', 'unit', 'lifestage'), "_") |>
+  glimpse()
+
+
+# update DSMhabitat values ------------------------------------------------
+
+watersheds <- unique(all_hab_data_long$watershed)
+watersheds_trunc <- watersheds[!(watersheds %in%  c('North Delta', "South Delta"))]
+
+## spawning ----------------------------------------------------------------
+
+r_to_r_tmh_sr_spawn <- DSMhabitat::sr_spawn$biop_itp_2018_2019
```

Comment:
It would be great if you could functionalize this. Then you don't have to run separately for each run

Reply:
Yes, agreed that it will make it better - will add to the list.